### ISSUE TITLE
Creation of an official Discord server for the Node.js project

### ISSUE THREAD

**Issue by ovflowd** [nodejs/admin#872]
Discord is a massive hub for collaboration, communication, and projects to reach their audiences.

It's becoming trendy for massive open-source projects and tech companies to have Discord Servers (ie, Cloudflare, Sentry, GitHub Education, etc.)

The upcoming [Ambassador Program](https://github.com/mhdawson/io.js/blob/ambassador-program/doc/contributing/advocacy-ambasador-program.md) gives us even more reason for allowing users to interact with Node.js ambassadors and take a closer part of the community.

### The goals of such a Discord server are simple:

- Yet another space for sharing announcements (mirror of RSS feeds and other sort of announcements)
- A space for Node.js enthusiasts to talk and discuss together
  - GitHub Discussions never worked out, and usually, we see Node.js technical discussions happening in places such as Twitter/X, Stackoverflow, and others, which happen not to be the most ideal form of asynchronous communication
- Although we have a Slack instance (community governed), rarely anyone joins or communicates there (the same with the IRC server)
  - With the Discord server, we could officially shut down both our IRC nodes, which have been moderated/managed by a single person (@ljharb), and the same applies to the Slack instance.
  - It also allows the OpenJS Foundation Slack to focus on OpenJS-related discussions and communication for existing Collaborators (more internal conversation rather than community-facing)
- The appeal for users to use Discord is way higher, due to:
  - One account for all your Discord Servers
  - Way better chatting, communication, and sharing features
  - Numerous integrations that could leverage our experience and collaboration with our Community
  - Way better Moderation tools
- With the Discord server, we could easily have channels that vary from:
  - Feature Feedbacks
  - Support Threads (which could allow Node.js developers to help each other and quickly find existing/prior questions)
  - Run Community Surveys/Polls with ease and way better granularity and reach (since many people don't use X/Twitter anymore)
  - Host streams from Node.js Collaborators / Ambassadors
- The instance could be moderated by the Moderation Team + Volunteers and managed by the TSC + trusted volunteers

Running a Discord server could be better for our community and our ability to reach the outer layers of our community.

**Comment by targos** [comment]
Previous discussion: https://github.com/nodejs/admin/issues/53

**Comment by ovflowd** [comment]
> Previous discussion: #53

Cool. Also the state of Discord and its features today vs 2018 has been very different.

**Comment by RedYetiDev** [comment]
> The instance could be moderated by the Moderation Team + Volunteers and managed by the TSC + trusted volunteers

I'm sure the triage team could also pitch in if needed! I'm +1 on the incorperation of a Discord server.

Not to rush things, but if we were to do this, we should probably reach out to Discord's legal team and get `https://discord.com/invite/nodejs`

**Comment by richardlau** [comment]
FWIW I'd be unlikely to move over to Discord -- it would be an additional communication platform and I wouldn't create a new account there (or anywhere else) just for Node.js.

**Comment by ovflowd** [comment]
> FWIW I'd be unlikely to move over to Discord -- it would be an additional communication platform and I wouldn't create a new account there (or anywhere else) just for Node.js.

No one is forcing you there 🙏; You don't need to be there if you don't want to, and I agree yet another platform, but that would be for the community 👀

We definitely don't need all collaborators to be there. And to be honest this could be a pilot program to see how it succeeds.

**Comment by ovflowd** [comment]
> The third "goal" mentions potentially shutting down Slack, or is that specifically the non-OpenJS Foundation one?

It was an idea to slowly phase out the "Node Slackers" slack server that Jordan maintains by himself, but he said that he is not interested in shutting in down so that's fine 🤷

But we would be able to let the OppenJS Foundation Slack at least for Node.js to be more for collaborators and contributors and less for asking support/help. Which it is also not used much for that, but still some people ask support there.

**Comment by GeoffreyBooth** [comment]
This doesn't sound like an improvement to me. We already have the official Slack workspace, and GitHub of course. Adding another official channel means yet another place to check.

**Comment by mcollina** [comment]
Growing a community on Discord is _useful_. The official OpenJS _slack_ is for our communication, while Discord would be for end users.

I think it would become quite popular, too. It's also a great marketing channel, and what I would set up today instead of the unofficial Node.js slack.

The fundamental question is who will moderate this and provide Node.js knowledge to the users.

**Comment by targos** [comment]
I agree that growing a community is useful. There are probably already existing communities, though, so we would kind of be in competition with them.
As a user, if I found and invested myself in an existing Node.js-focused Discord community, with well-defined rules and many moderators and helpers, I would need a very convincing argument to switch to an "official" one (what does the official stamp change?).

**Comment by ovflowd** [comment]
> This doesn't sound like an improvement to me. We already have the official Slack workspace, and GitHub of course. Adding another official channel means yet another place to check.

Each one has different roles. It's agreeable that we see users asking/seeking help on the help repository and other platforms, including OpenJS Slack. The idea is that people will naturally switch over to Discord over time. It's ridiculous how popular Discord is nowadays for hosting tech communities.

I still believe many users would use our help repository, but I can see many of them going over Discord.

**Comment by ovflowd** [comment]
> As a user, if I found and invested myself in an existing Node.js-focused Discord community, with well-defined rules and many moderators and helpers, I would need a very convincing argument to switch to an "official" one (what does the official stamp change?).

For better or for worse, I haven't found any big community. That said, I bet there might be smaller-ish multi-purpose communities focusing on Node or more considerable general-purpose programming Discord servers with channels for Node.js.

The main difference between the official one and the other would be that it's an official community consisting of collaborators, ambassadors, streams, announcements, etc. People tend to go to official communities because they are official.

Since there's no official JavaScript Discord server, I can see people with JavaScript interests going over our Discord. After all, it is also the most popular language (JavaScript) among teenagers, young adults, and even folks that code Discord Bots 🤷

**Comment by ovflowd** [comment]
There are also benefits in being an official community, as Discord gives some perks to Community servers.

**Comment by RomainLanz** [comment]
I am all in for that!

**Comment by tpoisseau** [comment]
As an early adopter of Discord, I appreciate its use for small communities. However, I'm wary of the future of the platform, which is funded by venture capitalists. We're starting to see more and more attempts at return on investment at the expense of the user experience. Even if it's extremely popular today, I wonder about its long-term future.

As mentioned by @targos, there is already a well-structured Node.js Discord server, perhaps it would be interesting to discuss with them how to define it as an official Discord in order to avoid creating a new community, dividing instead of bringing together.

To prepare for the creation or investment of a large community, I think it's very important to think seriously about moderation (who's in charge, the time that can be allocated to it, the rules). You also need to define the aims and intentions of such a community, whether it's to :

- offer a place where people can help each other (could limit out-of-context issues in Node.js repos)
- be a place for more informal communication between Node.js developers/maintainers and runtime users, in order to determine roadmap elements or prioritise improvements.
- transmit announcements (releases)
- promote the runtime. (marketing)
- survey the community

How to ensure that the community is attractive and engaged, so that what happens there can be followed without being drowned out.
Managing a large community is an important and time-consuming task. Are some Node.js staff willing and able to get involved?

**Comment by ovflowd** [comment]
> As mentioned by @targos, there is already a well-structured Node.js Discord server, perhaps it would be interesting to discuss with them how to define it as an official Discord in order to avoid creating a new community, dividing instead of bringing together.

That is also a valid alternative :) but we would somewhat need to claim ownership of that server. I'm all fine with allowing the current people managing it to also keep managing it, but there would definitely need to be changes 😅

**Comment by ovflowd** [comment]
> To prepare for the creation or investment of a large community, I think it's very important to think seriously about moderation (who's in charge, the time that can be allocated to it, the rules). You also need to define the aims and intentions of such a community, whether it's to :

We have experience managing communities. (Not to mention we do have members of the Node.js community with real community management experience, myself included)

We understand that this is something complex and that requires care/attention, so we wouldn't do this on a whim.

> Managing a large community is an important and time-consuming task. Are some Node.js staff willing and able to get involved?

We don't have Staff. But we would have volunteers to manage it.

>  You also need to define the aims and intentions of such a community, whether it's to :

This is all written on my opening post 👀 and I agree with all these items.

**Comment by tpoisseau** [comment]
I may have forgotten this part while writing my post, sorry 😅.

By ‘Staff’, I didn't mean employees but rather something like active, permanent members who take care of managing the Node.js project.
I'm not entirely familiar with the governance of the project, but I suppose such people exist.

**Comment by RedYetiDev** [comment]
> I'm not entirely familiar with the governance of the project, but I suppose such people exist. 

FWIW https://github.com/nodejs/node/blob/main/GOVERNANCE.md

**Comment by ovflowd** [comment]
Heyo 👋 I've had a nice chat with @vcarl which is the admin of the Nodeiflux Discord Server, and we had some really cool collaboration happening there ✨ 

I'd say they are inclined into collaborating with us to transform their server into Node's official Discord Server, but the caveat is they definitely need to be part of the process (which I 100% agree, otherwise it'd sound as a hostile takeover)

I believe there's lot we can gain and experiences that can be shared on how we can shape such a community into a successful place for Node's community :)

**Since this requires TSC input, I want a vote for:**

- The TSC delegates me as the DRI (Direct Responsible Individual / Champion) of this initiative, which allows me to:
  - Be the leader of this initiative (by taking care of organising and ensuring this initiative succeeds)
  - Responsible for communicating decisions and progresses with the TSC in a weekly basis (or bi-weekly as soon as I have updates)
  - Serve as an intermediate to make decisions regarding the adoption of Discord as a platform
    - All major decisions will first be communicated to the TSC, and the TSC will decide to approve (consensus seeking) or block such decisions.
- This is not a vote for the TSC to decide that we will make an official Discord server but a vote that such an initiative can be kicked off and that work can happen behind the scenes. This is an experiment, but we want to make it **right** and shape it into something we're all happy about before formalizing it and making it official.
- This vote also means the upcoming work should respect and abide by the OpenJS Foundation's CoC / Node.js CoC and governance model.
- I'll also involve the foundation for related marketing, legal, community, and related discussions, as they are the ideal government body to support us.

Nothing must get released/published without the explicit approval of the TSC and explicit 👍 of the Foundation regarding our being 👍 in legal, branding, and marketing matters.

Please react with 👍 or 👎 (and if possible, why if a 👎 so we can address your concerns) regarding whether you're in favor or against the proposal above.

cc @nodejs/tsc (I'm also adding to the TSC agenda for visibility and if discussions need to be done)

**Comment by benjamingr** [comment]
@ovflowd I like your idea of endorsing/helping existing communities and helping them grow. I'm wondering if we can keep the current "management" and let them moderate and be involved in a less direct capacity e.g.: 
 - Doing AMAs on endorsed/supported community servers.
 - Giving official announcements on new features.
 - Mentorship hours for new/junior developers and developers from underprivileged communities.
 - Code & Learns (like mentorship but for Node.js core contribution rather than general node help).

We can call it "officially endorsed but not official" and link to it in the website or something similar.

> This vote also means the upcoming work should respect and abide by the OpenJS Foundation's CoC / Node.js CoC and governance model.

My primary concern with an official Node.js discord server is (as usual) this . The OpenJS slack server is currently in a pretty bad state IMO and for me to be comfortable with Discord I would need (several) people to be willing to moderate it consistently.                                                                                      

> I don't mind if we ask for volunteers to set it up (I wouldn't mind at all). Still, ultimately, I believe the Moderation team would be responsible for moderating it + some volunteers. We can draft something more concrete. There's some energy lift in setting it up.

The moderation team is ~4-5 active people, even if we get to 10 active people I'm not sure most of them would be interested or have capacity in moderating another space but I'm not sure since discussions about slack and other spaces were a long while ago.

That said, if the moderation team agrees I think it would be an effective body in enforcing the CoC.

**Comment by ovflowd** [comment]
>  I'm wondering if we can keep the current "management" and let them moderate and be involved in a less direct capacity e.g.:

That's definitely what I was going after. But we should also allow the Node Moderation team to have elevated privileges there so that we can enforce moderation in case we need to.

> Doing AMAs on endorsed/supported community servers.
Giving official announcements on new features.
Mentorship hours for new/junior developers and developers from underprivileged communities.
Code & Learns (like mentorship but for Node.js core contribution rather than general node help).

I love this!

> My primary concern with an official Node.js discord server is (as usual) this . The OpenJS slack server is currently in a pretty bad state IMO and for me to be comfortable with Discord I would need (several) people to be willing to moderate it consistently.

I believe Nodeiflux had a great track of moderating their community well. I believe a pilot can be done. I also believe we can have volunteers to moderate there, which will have moderating powers there and only there.

> The moderation team is ~4-5 active people, even if we get to 10 active people I'm not sure most of them would be interested or have capacity in moderating another space but I'm not sure since discussions about slack and other spaces were a long while ago.

I hear you, and I could help being that bridge. I'm pretty sure @vcarl would do their best to enforce our CoC policy there if they agree with adopting our CoC policy, which is the Contributor Covenant 2.0. As an official server, it also means it lies under the Foundation, but I believe we can delegate moderation to their team. We do not moderate the OpenJS Slack, although it is an official Node.js space.

The biggest +1 of having it as an official server is Discord itself can promote and showcase the official community server, and by being verified, there are extra perks. + I'm pretty sure more people will feel compelled to join it as an official.

**Comment by benjamingr** [comment]
>  We do not moderate the OpenJS Slack, although it is an official Node.js space.

Not that I'm aware (it being an official space)

**Comment by ovflowd** [comment]
> > We do not moderate the OpenJS Slack, although it is an official Node.js space.
> 
> Not that I'm aware (it being an official space)

🤔

**Comment by tniessen** [comment]
> With the Discord server, we could officially shut down both our IRC nodes

Where's this implication coming from? Is the assumption that users who currently use IRC would definitely want to migrate to Discord? Are we maintaining IRC right now because we _don't_ have a Discord community?

> It also allows the OpenJS Foundation Slack to focus on OpenJS-related discussions and communication for existing Collaborators (more internal conversation rather than community-facing)

FWIW, monitoring the various GitHub repositories _and_ the OpenJS Slack is already way more than I, and perhaps most collaborators, can handle. If the goal is to shift _some_ traffic from OpenJS Slack to Discord, that's fine by me, but if it doesn't fully replace an existing channel, I wouldn't expect any collaborator to engage with and/or monitor yet another channel.

> The biggest +1 of having it as an official server is Discord itself can promote and showcase the official community server, and by being verified, there are extra perks.

If the main aspect of having an "official" (or "officially endorsed but not official") server is just labeling it as such while not requiring us collaborators to ever engage there, I don't see a problem with this as long as those acting upon this initiative don't misrepresent what the Discord server is and, most importantly, moderate interactions there appropriately. As @benjamingr hinted, we certainly can't expect the existing moderation team to take on another space.

**Comment by ovflowd** [comment]
> Where's this implication coming from? Is the assumption that users who currently use IRC would definitely want to migrate to Discord? Are we maintaining IRC right now because we don't have a Discord community?

We don't maintain anything, we don't have an official IRC server. It's not an implication, just a suggestion based on the assumption that there are fewer users on IRC.

> FWIW, monitoring the various GitHub repositories and the OpenJS Slack is already way more than I, and perhaps most collaborators, can handle. If the goal is to shift some traffic from OpenJS Slack to Discord, that's fine by me, but if it doesn't fully replace an existing channel, I wouldn't expect any collaborator to engage with and/or monitor yet another channel.

My assumption is that you shouldn't monitor Discord. It's a place to hang out. And I'd say we would redirect community users to Discord instead of Slack.

> If the main aspect of having an "official" (or "officially endorsed but not official") server is just labeling it as such while not requiring us collaborators to ever engage there, I don't see a problem with this as long as those acting upon this initiative don't misrepresent what the Discord server is and, most importantly, moderate interactions there appropriately. As @benjamingr hinted, we certainly can't expect the existing moderation team to take on another space.

I'm 100% fine with that.

**Comment by tniessen** [comment]
Sounds good, thank you for clarifying @ovflowd :)

**Comment by ovflowd** [comment]
> Sounds good, thank you for clarifying @ovflowd :)

Of course! And sorry if my replies aren't the best. But I want to make sure I'mn very flexible to accommodate the needs of the TSC, and I bet we can find something that is close-to-ideal for all parties.

**Comment by vcarl** [comment]
Hey thanks for the ping, very much enjoyed our chat @ovflowd! Just catching up on the convo, I'll chime in a bit as a potential outside collaborator in executing this 😄 

To introduce myself, I'm Carl (usually vcarl on the internet) :wave: I've been helping oversee Reactiflux, currently hitting ~60-70k visitors/mo, since 2016 or so. I spun up Nodeiflux as a companion community for the JS backend around 2018 (remember when there weren't other runtimes??), and have kept it humming more or less solo since then. It's never been a major focus, but it's always had consistent activity so I never pulled the plug either. 

> I'd say they are inclined into collaborating with us to transform their server[…], but the caveat is they definitely need to be part of the process

💯❤️ Happy to be a collaborator here, my ask would be to take my ~8 years of overseeing online communities as a valuable perspective 🙂 I've taken my role in communities seriously, which has meant encountering a long list of unusually common scenarios.

> I'm wondering if we can keep the current "management" and let them moderate and be involved in a less direct capacity

Happy to be a primary point of contact, with moderation of the space happening on my end and by people I oversee but accountable to the Node org in some way. I was a significant driver of recruiting and establishing moderation norms in Reactiflux, I'm quite comfortable managing a growing community on my own and would find some added support structures for branding, news, and events to be super helpful. My concern would be with onerous bureaucracy and reporting, but that's certainly not inevitable.

> we should also allow the Node Moderation team to have elevated privileges there so that we can enforce moderation in case we need to.
> "officially endorsed but not official"
> As benjamingr hinted, we certainly can't expect the existing moderation team to take on another space.

Details here are definitely workable imo. We have an existing admin/moderator role system, and could very easily add an elevated moderator role for someone in an accountability/oversight role from the Node org.

> I'm pretty sure @vcarl would do their best to enforce our CoC policy there if they agree with adopting our CoC policy, which is the Contributor Covenant 2.0.

We based our current code of conduct off the Rust CoC, which I've absolutely done my best to enforce. The realities of an online chat system are that _mitigating harm_ is much harder to do than enforce the CoC, but I've done what I can. I also have a collaborator in Europe I trust to help from another timezone, and a number of helpful regulars who reliably draw my attention to the most important matters and react appropriately in the meantime. It's been a small but functional ecosystem for a few years now.

**Comment by mcollina** [comment]
I'm generically in favor of this proposal as long as it does not strain the current moderation team and a few people from us get admin rights.

From a practical/admin perspective we would need to charter a working group and give them autonomy in how they run the Discord server. That repo/wg would be the point of contact with the rest of the TSC.

Ultimately, this would be that the TSC is responsible for the space.

**Comment by tniessen** [comment]
> Ultimately, this would be that the TSC is responsible for the space.

What does "responsible for the space" mean? Is this another instance of "the TSC steps in as the last resort?"

**Comment by ovflowd** [comment]
I believe we have an initial agreement here. Can we get this discussed by the TSC?

Also cc @mhdawson since I wanted to involve the ambassador initiative in this

**Comment by mhdawson** [comment]
@ovflowd -> since I wanted to involve the ambassador initiative in this. Do you want to get together or did you have specific things you wanted to add to the program description in https://github.com/nodejs/node/blob/main/doc/contributing/advocacy-ambasador-program.md ?

**Comment by mhdawson** [comment]
@ovflowd do you want to come to the TSC meeting to present/discuss the initial agreement?  Reading through the thread I can guess what that is but and not 100% sure I would get it right.

**Comment by ovflowd** [comment]
> @ovflowd -> since I wanted to involve the ambassador initiative in this. Do you want to get together or did you have specific things you wanted to add to the program description in [nodejs/node@`main`/doc/contributing/advocacy-ambasador-program.md](https://github.com/nodejs/node/blob/main/doc/contributing/advocacy-ambasador-program.md?rgh-link-date=2024-06-17T16%3A03%3A51Z) ?

I think the only relevant pieces are that Ambassadors would have a role on the Discord, be there at their criteria/convenience and be able to host streams + promote their materials if wanted. Definitely something we can fine tune :) (Im just throwing some random ideas)

**Comment by ovflowd** [comment]
> @ovflowd do you want to come to the TSC meeting to present/discuss the initial agreement? Reading through the thread I can guess what that is but and not 100% sure I would get it right.

Sure, Id be more than happy to jump at the next TSC meeting to chatter about this.

**Comment by tniessen** [comment]
@ovflowd We briefly discussed during the TSC meeting.

> The biggest +1 of having it as an official server is Discord itself can promote and showcase the official community server, and by being verified, there are extra perks.

Our understanding is that marking the existing server as "official" would primarily help the existing Discord server in the ways you described and perhaps make it easier for people to find a Node.js-related Discord community. Secondarily, it would allow us to share announcements in some official capacity (as opposed to engaging with the Discord community like before). This sounds like it's good for the community overall, and as long as this doesn't require any active involvement of Node.js personnel, there were no major concerns.

We had one small (and hopefully, purely hypothetical) concern. Is this _marking it as official_ something that can be undone, if necessary? While I am sure this won't happen, if the Discord server does not end up living up to our community standards, would we be able to sever ties without forcing the server to shut down entirely?

**Comment by ovflowd** [comment]
> We had one small (and hopefully, purely hypothetical) concern. Is this _marking it as official_ something that can be undone, if necessary? While I am sure this won't happen, if the Discord server does not end up living up to our community standards, would we be able to sever ties without forcing the server to shut down entirely?

Yes, we can always reach out to Discord support and revoke the verification (server verification, which marks it as official)

**Comment by RedYetiDev** [comment]
Great! I'm glad this is coming together!

I look forward to the beginning of this server!

**Comment by ovflowd** [comment]
Alrighty, @nodejs/tsc let me know if we can start pushing this forward :)

**Comment by ovflowd** [comment]
> @ovflowd you need to say something like "if there are no remaining concerns, I'll go ahead and do _such and such_ on _such or such day_" if you want this to move forward.

Oh yeah, let me rephrase:

If there are no remaining concerns, I'll proceed with my plan from here: https://github.com/nodejs/admin/issues/872#issuecomment-2156422974

More specifically:

- Starting collaborating with @vcarl on the possible (if needed) restructure of certain aspects of the Server (such as rules, channels, roles, bots, moderation pieces etc)

There's nothing set in stone yet; hence, I will report back on updates and if other things are needed from the TSC side, and how things progress.

**Comment by GeoffreyBooth** [comment]
> If there are no remaining concerns

I don’t think the concerns from https://github.com/nodejs/admin/issues/872#issuecomment-2156428381 have been addressed yet? Namely, what the plan is for moderation of the new space and whether @nodejs/moderation has the capacity to take this on.

**Comment by bmuenzenmeyer** [comment]
> That said, if the moderation team agrees I think it would be an effective body in enforcing the CoC.

> I don’t think the concerns from [#872 (comment)](https://github.com/nodejs/admin/issues/872#issuecomment-2156428381) have been addressed yet? Namely, what the plan is for moderation of the new space and whether @nodejs/moderation has the capacity to take this on.

As a new member of the moderation team, I doubt I'll have dedicated time to monitor a Discord.

**Comment by ovflowd** [comment]
> > If there are no remaining concerns
> 
> I don’t think the concerns from [#872 (comment)](https://github.com/nodejs/admin/issues/872#issuecomment-2156428381) have been addressed yet? Namely, what the plan is for moderation of the new space and whether @nodejs/moderation has the capacity to take this on.

We did address those concerns, yes. Pretty much the plan is (as stated in previous comments):

- Some TSC/Moderation team members get moderation privileges on the Discord server
- The current moderators/staff of the discord server keep also enforcing moderation there (we can add extra volunteers only to moderate there)
- Our Moderation team is not required to moderate that space but can do so if they wish
- Discord Server moderators have a Slack channel with us (moderation team) so that we can quickly sync in case of emergencies and for weekly syncs from their side (maybe?)

**Comment by RedYetiDev** [comment]
Hi everyone,

I have a few clarifying questions:

- Currently, the project use Slack and GitHub Discussions for our topics. However, I find GitHub Discussions can be challenging because conversations often go off-topic, and resolved issues remain open. I think it might be beneficial to direct users to ask their questions on Discord instead. This could help common issues gain more visibility and traction. Would this be acceptable?

- Additionally, if the project starts using Discord for user questions, I believe the triage team should be a part of answering them. While we don't need special permissions, I think it would be helpful to have the same recognition as GitHub Member badge shows. Demonstrating credibility is crucial for effectively resolving issues, and this recognition could aid in resolving issues more efficiently and accurately.

**Comment by ovflowd** [comment]
> Additionally, if the project starts using Discord for user questions, I believe the triage team should be a part of answering them. While we don't need special permissions, I think it would be helpful to have the same recognition as GitHub Member badge shows. Demonstrating credibility is crucial for effectively resolving issues, and this recognition could aid in resolving issues more efficiently and accurately.

I believe Node.js Collaborators could be given a "Node.js Collaborator" role and triagers a "Node.js Triager" role. This was already something I wanted to do.

> Additionally, if the project starts using Discord for user questions, I believe the triage team should be a part of answering them. While we don't need special permissions, I think it would be helpful to have the same recognition as GitHub Member badge shows. Demonstrating credibility is crucial for effectively resolving issues, and this recognition could aid in resolving issues more efficiently and accurately.

I'm not sure about that. Although I believe we could disable GitHub Discussions as they are rarely monitored/viewed and moderated anyways. Discord has a better platform for making Discussion/Help threads. cc @nodejs/tsc

**Comment by tniessen** [comment]
> I think it might be beneficial to direct users to ask their questions on Discord instead (Not a replacement, but a "hey, we also have Discord, you can ask questions there, any they might gain more traction"). This could help common issues gain more visibility and traction. Would this be acceptable?

An upside of using GitHub is that org members can ping collaborators or teams when it seems appropriate, and we can also get folks from libuv and V8 involved rather easily. Some of the most knowledgeable contributors almost exclusively interact through GitHub, so directing users away from GitHub might be directing users away from those.

**Comment by RedYetiDev** [comment]
FWIW I was specifically referring to GitHub Discussions

**Comment by tniessen** [comment]
My response also applies to GitHub Discussions :)

**Comment by ovflowd** [comment]
> Some of the most knowledgeable contributors almost exclusively interact through GitHub, so directing users away from GitHub might be directing users away from those.

But I assume that Discussions was used for general chat. I assume help requests can still be directed to node/help and feature requests/bugs to node/node

So I assume, it would still make sense to disable Discussions, maybe? IDK :shrug: I'm fine either ways, but I'm fond of reducing the number of parallel platforms we have for people to write something. And if GitHub Discussions are mostly used for general discussions, Discord is an ideal place... I know there's a tradeoff of "but it's all on GitHub, no extra account needed", and I'd say yeah. I think in the end it boils down to, "do we really want to have GitHub Discussions?" has anything important happened there for the last X period of time?

**Comment by mhdawson** [comment]
From the discussion so far, my understanding was that the Discord effort was around providing support for those that already use Discord as their communication channel. I don't think we should anything that would require Node.js collaborators to join Discord in order to be involved in day-to-day work of the project.

I guess I'm just agreeing with what @tniessen said in part.  For any discussion where we want project collaborators to be involved, we should not push those discussions to Discord.

**Comment by ovflowd** [comment]
> From the discussion so far, my understanding was that the Discord effort was around providing support for those that already use Discord as their communication channel. I don't think we should anything that would require Node.js collaborators to join Discord in order to be involved in day-to-day work of the project.
> 
> I guess I'm just agreeing with what @tniessen said in part. For any discussion where we want project collaborators to be involved, we should not push those discussions to Discord.

That is a correct assessment. My point was unrelated (somewhat) and more an open question: Do we need GitHub discussions? Are we using them? Do they serve for some specific purpose?

And no, we won't require collaborators to join Discord although they are more than welcome and invited to join :)

**Comment by ovflowd** [comment]
FYI, we've started drafting a temporary "Beta" server to use as a template and then apply to the existing Discord server.

I'll follow-up with more updates once I have.

**Comment by ovflowd** [comment]
As many have noticed, we released the Discord server! https://nodejs.org/en/blog/announcements/official-discord-launch-announcement

Here is also a link with follow-up tasks: https://docs.google.com/document/d/1ED-P6vAUswelxqvlKE3iZ-NPHnBZpVP3I8vbygPnGMI/edit?usp=sharing
